scripts/uosc/elements/Menu.lua

```diff
@@ -698,13 +722,20 @@ function Menu:search_query_update(query)
 	request_render()
 end
 
-function Menu:search_backspace()
-	local pos, search_text = #self.current.search.query - 1, self.current.search.query
-	while pos > 1 and search_text:byte(pos) >= 0x80 and search_text:byte(pos) <= 0xbf do
+---@param event? string
+function Menu:search_backspace(event)
+	local pos, old_query, is_palette = #self.current.search.query - 1, self.current.search.query, self.current.palette
+	while pos > 1 and old_query:byte(pos) >= 0x80 and old_query:byte(pos) <= 0xbf do
```

Comment:
I should have left a comment here that the while loop is for skipping utf8 continuation bytes.